drivers/nrf_802154_sl_opensource/src/nrf_802154_sl_rsch.c

```diff
@@ -95,14 +95,7 @@ bool nrf_802154_rsch_timeslot_is_requested(void)
 
 bool nrf_802154_rsch_prec_is_approved(rsch_prec_t prec, rsch_prio_t prio)
 {
-    if (prec == RSCH_PREC_HFCLK)
-    {
-        return prio == RSCH_PRIO_IDLE ? true : m_ready;
-    }
-    else
-    {
-        return true;
-    }
+    return prio == RSCH_PRIO_IDLE ? true : m_ready;
```

Comment:
This change looks like reintroducing a bug that was recently fixed.

Is this described as fix 4?

> The issue which manifests itself in disabled NVIC interrupts in SED roles.

Could you comment what's the purpose of this change and how is the problem fixed now?